### PR TITLE
[browser][wasm] Fix assembly name not found due to rename of assembly.

### LINKED ISSUE #41036: [browser][wasm] Marshal tests "Could not find assembly"
The following error is now thrown in the tests due to PR: https://github.com/dotnet/runtime/pull/40478

```
Error: Could not find assembly: System.Runtime.InteropServices.JavaScript.Tests
```

### PR DESCRIPTION
**Pull request by kjpou1**
Due to the renaming of the assembly System.Runtime.InteropServices.JavaScript to System.Private.Runtime.InteropServices.JavaScript the marshal tests will fail once they are turned back on.

This PR does not reactivate the tests only fixes a problem that will be presented in the future if someone is working on those tests in the future.

The active issue that turns off those tests is https://github.com/dotnet/runtime/issues/40112.  Once this issue is addressed then the tests can be turned back on.

Closes https://github.com/dotnet/runtime/issues/41036

### CODE REVIEW

**Comment by SamMonoRT** [comment]
Please confirm if these tests are now passing in the CI checks after this change prior to merging.

**Review by SamMonoRT — APPROVED** [review]

**Comment by kjpou1** [comment]
This just changes the assembly name and not turning the tests on.  There is still work being done to ensure these tests are passing in other PR's.

**Comment by SamMonoRT** [comment]
LGTM if enabling the tests is done via a separate PR. Add a note regarding that PR/issue details or link that issue here for visibility.

**Comment by kjpou1** [comment]
Here is the active issue that turns the tests off.  https://github.com/dotnet/runtime/issues/40112

Updated the main issue comments that this does not turn the tests back on and will be handled later.

**Review by SamMonoRT — APPROVED** [review]
Good to merge.

**Pull request merged**
